plugins/module_utils/template.py

```diff
@@ -0,0 +1,336 @@
+            variable_start_string=variable_start_string,
+            variable_end_string=variable_end_string,
+            comment_start_string=comment_start_string,
+            comment_end_string=comment_end_string,
+            line_statement_prefix=line_statement_prefix,
+            line_comment_prefix=line_comment_prefix,
+            trim_blocks=trim_blocks,
+            lstrip_blocks=lstrip_blocks,
+            newline_sequence=newline_sequence,
+            keep_trailing_newline=keep_trailing_newline,
+            loader=jinja2.FileSystemLoader(
+                searchpath=template_dir,
+                encoding=encoding,
+            ),
+            auto_reload=auto_reload,
+            autoescape=True,
+        )
+
+    def render_file_template(self, file_path, variables):
+        """Loads a template from the templates directory and renders
+        it using the Jinja2 environment configured in the object.
+
+        Arguments:
+            file_path (str): Relative path (from the template directory)
+                    to a template.
+            variables (dict): Dictionary containing the variables and
+                    their values that will be substituted in the template.
+
+        Returns:
+            tuple -- Filepath to a temporary directory that contains the
+                    rendered template, and the complete filepath to the
+                    rendered template.
+
+        Raises:
+            TemplateNotFound: When the template file doesn't exist in the
+                    template directory.
+            TemplateError: When rendering of the template fails.
+            FileExistsError: When there is an error while trying to create the
+                    temp directory for rendered templates.
+            PermissionError: When there is an error accessing the temp directory.
+            IOError: When there is an error writing the rendered template.
+            ValueError: When there is an error writing the rendered template.
+        """
+        try:
+            template = self.templating_env.get_template(file_path)
+            rendered_contents = template.render(variables)
+        except jinja2.TemplateNotFound as err:
+            raise jinja2.TemplateNotFound("Template {0} was not found: {1}".format(
+                file_path,
+                to_native(err)
+            ))
+        except jinja2.TemplateError as err:
+            raise jinja2.TemplateError("Error while rendering {0}: {1}".format(
+                file_path,
+                to_native(err)
+            ))
+
+        try:
+            temp_template_dir = tempfile.mkdtemp()
+        except FileExistsError as err:
+            raise FileExistsError("Unable to create directory for rendered templates: {0}".format(
+                to_native(err)
+            ))
+        except PermissionError as err:
+            raise PermissionError("Error while trying to access temp directory for templates: {0}".format(
```

Comment:
Very small, is it `temp` , `tmp` just play it safe and say temporary. Python does have precedence rules for how it finds a temporary directory to write into so its not always /tmp because i have had to instruct users on how to override this so that our modules an write into a bigger directory, basically saying maybe play it safe and if it makes sense, just say 'temporary'